eth/protocols/eth/peer.go

```diff
@@ -369,6 +400,39 @@ func (p *Peer) AsyncSendNewBlock(block *types.Block, td *big.Int) {
 	}
 }
 
+// SendVotes propagates a batch of votes to the remote peer.
+func (p *Peer) SendVotes(votes []*types.VoteEnvelope) error {
+	// Mark all the transactions as known, but ensure we don't overflow our limits
+	for p.knownVotes.Cardinality() > max(0, maxKnownTxs-len(votes)) {
+		p.knownVotes.Pop()
```

Comment:
To avoid uint64 overflow, better to use p.knownVotes.Cardinality() + len(votes) <  maxKnownTxs-len(votes)